back/app/controllers/web_api/v1/projects_controller.rb

```diff
@@ -52,6 +52,74 @@ def index
     )
   end
 
+  # For use with 'Open to participation' homepage widget.
+  # Returns all published or archived projects that are visible to user
+  # and in an active participatory phase (where user can do something).
+  # Ordered by the end date of the current phase, soonest first (nulls last).
+  def index_projects_with_active_participatory_phase
+    # Projects user can see, with active participatory (not information) phase & include the phases.end_at column
+    # We could use the ProjectPermissionSevice step to filter for active phases, but doing it here is more efficient.
+    subquery = policy_scope(Project)
+      .joins('INNER JOIN admin_publications AS admin_publications ON admin_publications.publication_id = projects.id')
+      .where(admin_publications: { publication_status: 'published' })
+      .joins('INNER JOIN phases AS phases ON phases.project_id = projects.id')
+      .where(
+        'phases.start_at <= ? AND (phases.end_at >= ? OR phases.end_at IS NULL) AND phases.participation_method != ?',
+        Time.zone.now.to_fs(:db), Time.zone.now.to_fs(:db), 'information'
+      )
+      .select('projects.*, phases.end_at AS phase_end_at')
+
+    # Perform the SELECT DISTINCT on the outer query
+    @projects = Project
+      .from(subquery, :projects)
+      .distinct
+      .order('phase_end_at ASC NULLS LAST')
+      .preload(phases: { permissions: [:groups] })
+
+    # Projects user can participate in, or where such participation could (probably) be made possible by user
+    # (e.g. user not signed in).
+    # Unfortunately, this breaks the query chain, so we have to start a new one after this.
+    # Also, we will need the action descriptors again later, so we will store them.
+
+    # # Step 1: Create pairs of project ids and action descriptors.
+    # # Since this is the last filtering step, we will keep going
+    # # until we reach the limit required for pagination.
+    pagination_limit = calculate_pagination_limit
+
+    project_descriptor_pairs = {}
+    @projects.each do |project|
+      service = Permissions::ProjectPermissionsService.new(
+        project, current_user, user_requirements_service: user_requirements_service
+      )
+      action_descriptors = service.action_descriptors
+
+      if Permissions::ProjectPermissionsService.participation_possible?(action_descriptors)
+        project_descriptor_pairs[project.id] = action_descriptors
```

Comment:
Doesn't feel like we should need to get the action descriptors and then pass them back in - maybe been done for performance reasons? Would also need some refactoring of the service too, to get the following to work.
```suggestion
      if service.participation_possible?
        project_descriptor_pairs[project.id] = service.action_descriptors
```